Sources/CoreAardvark/Logging/ARKExceptionLogging.m

```diff
@@ -50,12 +51,25 @@ void ARKHandleUncaughtException(NSException *_Nonnull exception)
     NSLock *const logDistributorsLock = ARKGetUncaughtExceptionLogDistributorsLock();
     [logDistributorsLock lock];
 
+    dispatch_group_t observerGroup = dispatch_group_create();
+
     for (ARKLogDistributor *const logDistributor in ARKUncaughtExceptionLogDistributors) {
         [logDistributor logWithType:ARKLogTypeError userInfo:nil format:@"Uncaught exception '%@':\n%@", exception.name, exception.debugDescription];
         [logDistributor distributeAllPendingLogsWithCompletionHandler:^{}];
         [logDistributor waitUntilAllPendingLogsHaveBeenDistributed];
+
+        for (id<ARKLogObserver> observer in logDistributor.logObservers) {
+            if ([observer respondsToSelector:@selector(processAllPendingLogsWithCompletionHandler:)]) {
+                dispatch_group_enter(observerGroup);
+                [observer processAllPendingLogsWithCompletionHandler:^{
+                    dispatch_group_leave(observerGroup);
+                }];
+            }
+        }
     }
 
+    dispatch_group_wait(observerGroup, DISPATCH_TIME_FOREVER);
```

Comment:
Please add a good comment explaining why we're waiting and also why waiting here is safe. Waiting means future contributors are going to be a bit scared to touch this code, especially given that we're in the midst of a lock. Let's help them.

Reply:
Added a note about why we're waiting in 80045c8.

I don't have a great answer for why (or if) it's safe to wait here. As far as I know, this shouldn't make things worse, but at this point the app is already in an unstable state, so it's hard to know for sure. I changed it to 30 seconds and moved the wait until after we unlock to avoid any potential deadlock there.